cub/cub/detail/choose_offset.cuh

```diff
@@ -56,6 +56,36 @@ struct ChooseOffsetT
                                          unsigned long long>::type;
 };
 
+template <typename T,typename = void>
+struct is_iterator : std::false_type {};
+template <typename T>
+struct is_iterator<T,std::void_t<typename std::iterator_traits<T>::iterator_category>> : std::true_type {};
+template <typename T>
+inline constexpr bool is_iterator_v = is_iterator<T>::value;
```

Comment:
Inline variables are a C++17 feature, so we *should* guard this. Otherwise certain compilers will yell at you very loudly (couch gcc-6)

```suggestion
_LIBCUDACXX_INLINE_VAR constexpr bool is_iterator_v = is_iterator<T>::value;
```